Python 3.5 support?

If you have questions about a specific use case, or you are not sure whether this is a bug or not, please post it to our discourse channel: https://discourse.pymc.io  

## Description of your problem

Hi, is Python 3.5 supported anymore?

```python
/usr/local/envs/testenv_3.5_PYSTAN_latest_PYRO_latest_EMCEE_2/lib/python3.5/site-packages/_pytest/python.py:511: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
/usr/local/envs/testenv_3.5_PYSTAN_latest_PYRO_latest_EMCEE_2/lib/python3.5/site-packages/py/_path/local.py:701: in pyimport
    __import__(modname)
<frozen importlib._bootstrap>:968: in _find_and_load
    ???
<frozen importlib._bootstrap>:957: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:664: in _load_unlocked
    ???
<frozen importlib._bootstrap>:634: in _load_backward_compatible
    ???
/usr/local/envs/testenv_3.5_PYSTAN_latest_PYRO_latest_EMCEE_2/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:299: in load_module
    six.exec_(co, mod.__dict__)
arviz/tests/test_plots.py:8: in <module>
    import pymc3 as pm
/usr/local/envs/testenv_3.5_PYSTAN_latest_PYRO_latest_EMCEE_2/lib/python3.5/site-packages/pymc3/__init__.py:10: in <module>
    from .model_graph import model_to_graphviz
E     File "/usr/local/envs/testenv_3.5_PYSTAN_latest_PYRO_latest_EMCEE_2/lib/python3.5/site-packages/pymc3/model_graph.py", line 50
E       vars: MutableSet[RV] = set(self.var_list)
E           ^
E   SyntaxError: invalid syntax
```

**Please provide any additional information below.**
This happened after https://github.com/pymc-devs/pymc3/commit/9f3bba04b15b9e0b046e651b63b82b93146dcae3

## Versions and main components

* PyMC3 Version: 3.6 git Master
* Theano Version: 1.0.4
* Python Version: Python 3.5.6
* Operating system: ArviZ Docker (conda/miniconda3)
* How did you install PyMC3: pip

--- a/RELEASE-NOTES.md
+++ b/RELEASE-NOTES.md
@@ -12,7 +12,6 @@
 - Used `numpy.vectorize` in `distributions.distribution._compile_theano_function`. This enables `sample_prior_predictive` and `sample_posterior_predictive` to ask for tuples of samples instead of just integers. This fixes issue #3422.
 
 ### Maintenance
-- Fixed an issue in `model_graph` that caused construction of the graph of the model for rendering to hang: replaced a search over the powerset of the nodes with a breadth-first search over the nodes. Fix for #3458.
 - All occurances of `sd` as a parameter name have been renamed to `sigma`. `sd` will continue to function for backwards compatibility.
 - Made `BrokenPipeError` for parallel sampling more verbose on Windows.
 - Added the `broadcast_distribution_samples` function that helps broadcasting arrays of drawn samples, taking into account the requested `size` and the inferred distribution shape. This sometimes is needed by distributions that call several `rvs` separately within their `random` method, such as the `ZeroInflatedPoisson` (Fix issue #3310).
@@ -38,6 +37,8 @@
 - Fixed the `Multinomial.random` and `Multinomial.random_` methods to make them compatible with the new `generate_samples` function. In the process, a bug of the `Multinomial.random_` shape handling was discovered and fixed.
 - Fixed a defect found in `Bound.random` where the `point` dictionary was passed to `generate_samples` as an `arg` instead of in `not_broadcast_kwargs`.
 - Fixed a defect found in `Bound.random_` where `total_size` could end up as a `float64` instead of being an integer if given `size=tuple()`.
+- Fixed an issue in `model_graph` that caused construction of the graph of the model for rendering to hang: replaced a search over the powerset of the nodes with a breadth-first search over the nodes. Fix for #3458.
+- Removed variable annotations from `model_graph` but left type hints (Fix for #3465). This means that we support `python>=3.5.4`.
 
 ### Deprecations
 

--- a/pymc3/model_graph.py
+++ b/pymc3/model_graph.py
@@ -47,10 +47,10 @@ class ModelGraph:
         """
 
         # this contains all of the variables in the model EXCEPT var...
-        vars: MutableSet[RV] = set(self.var_list)
+        vars = set(self.var_list)
         vars.remove(var)
-        
-        blockers: MutableSet[RV] = set()
+
+        blockers = set()
         retval = set()
         def _expand(node) -> Optional[Iterator[Tensor]]:
             if node in blockers:

--- a/pymc3/model_graph.py
+++ b/pymc3/model_graph.py
@@ -1,4 +1,3 @@
-import itertools
 from collections import deque
 from typing import Iterator, Optional, MutableSet
 
@@ -12,17 +11,6 @@ import pymc3 as pm
 # this is a placeholder for a better characterization of the type
 # of variables in a model.
 RV = Tensor
-
-
-def powerset(iterable):
-    """All *nonempty* subsets of an iterable.
-
-    From itertools docs.
-
-    powerset([1,2,3]) --> (1,) (2,) (3,) (1,2) (1,3) (2,3) (1,2,3)
-    """
-    s = list(iterable)
-    return itertools.chain.from_iterable(itertools.combinations(s, r) for r in range(1, len(s)+1))
 
 
 class ModelGraph:

